### ISSUE TITLE
Ability to build single entry point only

### ISSUE THREAD

**Issue by Carlosamouco** [ng-packagr/ng-packagr#2732]
## Type of Issue

```
[ ] Bug Report
[X] Feature Request
```

## Description

When developing an Angular library, I want to be able to recompile only the entry point / sub entry point that I am working on. 

Currently, when I am building a library in watch mode, all the entry points that depend on the entry point that I am working on are recompiled even when I don't make any changes that affect the public API of that entry point. 

Even though this is the intended behaviour, it slows a lot the development processes when there are multiple dependent entry points that are being recompiled for no reason. Also, if I have an application consuming this library with live reloading through "ng serve", it ends up taking considerable time to apply the change due to all the recompilations of the entry points that happen in the library, and it makes testing the change very painful and slow.

#### How To Reproduce

Considering a package named "my-lib"
Have at least two entry points:
- my-lib
- my-lib/sub

1. Make the "my-lib" import the "my-lib/sub" entry point.
2. Change anything in "my-lib/sub".
3. Both entry points are always recompiled even when there isn't anything changing in the public API of the "my-lib/sub".

#### Expected Behaviour

Have a flag in the build command that allows us to define which entry points we want to watch for changes. For the first build, all the entry points should be compiled as usual.

ex: ng build my-lib --watch /sub

This would improve the development experience considerably.

**Comment by alan-agius4** [comment]
Duplicate of #925

**Comment by Carlosamouco** [comment]
I read the thread in #925. Is this still not planed? Is there any known tooling that I can use to achieve this or any way to customize the ng-packagr?

**Comment by alan-agius4** [comment]
Indeed, this is still not planned.

No, it is not possible to customize ng-packagr to build only a single entry-point.

**Comment by Carlosamouco** [comment]
Would you consider discussing how it should be implemented and accept a PR for this? Or do you think it does not make sense to include such feature in here?
Personally I think It would be a great improvement in the developer experience.

**Comment by Carlosamouco** [comment]
I think I should have provided more context on our use case.
Whenever we are working, in a component of a library, all the entry points that import that component appear to be rebuilding. It makes sense to typecheck the affected entry points but I see no reason to recompile them or emit declaration files since the code is unchanged. I may be mistaken but this is what appears to be happening. 

Since for example, in the @angular/material repository, each logical component is isolated in a entry point, we tried to transpose that into our libraries, but any rebuilds become really slow if we change a file in a entry point that is shared across multiple sub entry points.

Since almost all the time that we are working on a library, we have also the dev server from the "ng serve" running in order to test the changes, it makes all the typechecking that was performed by the ng-packagr redundant, since we also get typechecking from the "ng serve". And while all the affected libraries are being rebuilt, the dev server becomes "stuck" trying to rebuild the app. I am not sure why, since no code should have changed in the affected entry points.

At the end, we end up wasting more than a minute to test a simple change in one file, witch is painfully slow in my opinion.

I am not saying that watching a single entry point is the solution for this, but maybe it would be nice to have a way to disable/enable this behaviour when in watch mode. Maybe a new config in the ng-package.json that allows us to skip rebuilds for affected entry points  or be able to make those rebuilds less prompt to output changes that trigger app rebuilds would be great.

Please, can you tell me if it is more clear to you now and if this is worth opening a new issue?

**Comment by Carlosamouco** [comment]
I took a look at the code, and I think there are a couple of thing that would help allot when it comes to application rebuilding after  triggering a recompilation of a library.

From the tests that I preformed, there are a couple of things that are performed during the build of affected entry points that would be nice if they were skipped.

I think that, only the directly changed entry point should touch the version of the package.json file, rebuild and flatten the code. Other dependent entry points should just perform type checking and not emit anything to not trigger unnecessary rebuilding's of the application. It would also be nice to have a setting to skip type checking of affected libraries during development, when we are running the "ng serve" since type checking is also performed at the app level if we have "skipLibCheck" set to false.

Since I am diverging from my initial request, I will open a new issue request following this.
